src/states/joining_peer.rs

```diff
@@ -112,7 +112,11 @@ impl JoiningPeer {
     }
 
     fn send_join_requests(&mut self) {
-        let conn_infos = self.conn_infos.clone();
+        let conn_infos: Vec<_> = self
+            .p2p_nodes
+            .iter()
+            .map(|p2p_node| p2p_node.connection_info().clone())
+            .collect();
         for dst in conn_infos {
             info!("{} - Sending JoinRequest to {:?}", self, dst);
```

Comment:
It would be more useful to log the whole `P2pNode` here.

Reply:
Yep agree. I think this whole conn_info lookup will go away though in the upcoming PR